server/migrations/usersMigration.js

```diff
@@ -12,17 +12,20 @@ async function migrateUsers() {
       .collection('users')
       .find({})
       .forEach((user) => {
-        const userPreferences = calculatePreferences(user)
+        const { siteSubjects, updatedPreferences } = calculatePreferences(user)
 
         appDatabase.collection('users').updateOne(
           { _id: ObjectId(user._id) },
           {
             $set: {
-              preferences: userPreferences,
+              preferences: updatedPreferences,
+              siteSubjects,
             },
             //removes the field and the value from the document
             $unset: {
               account_expires: '',
+              'preferences.star': '',
+              'preferences.complete': '',
```

Comment:
do we need this if we're setting the preferences above? Also, what do you think about keeping the old preferences in a new attribute, like `legacyPreferences` in case our migration goes wrong we don't lose all that data. We can remove that field once we're confident the migration ran successfully.

Reply:
Thank you for that. I added the legacyPreferences property and it will store a JSON string.
We have to unset the `star` and `complete` properties otherwise prisma will throw errors since those properties are undefined in the users model

Reply:
https://github.com/AMP-SCZ/dpdash/pull/241/commits/bf0423c6c7f4d2b24d8c720cb39b4accdaccbbc7